R-package/R/utils.R

```diff
@@ -74,6 +74,33 @@ check.booster.params <- function(params, ...) {
         params[['monotone_constraints']] = vec2str
   }
   
+  # interaction constraints parser
+  if (!is.null(params[['int_constraints']])){
+    # check input class
+    temp.in_class <- unique(sapply(params[['int_constraints']], class))
+    if (length(temp.in_class) > 1) stop('invalid class specification for int_constraints')
+    if (is.null(column_names)) stop('require column names to set interaction constraints')
```

Comment:
@BlueTea88 Could we please keep things consistent, with the functionality primarily dependent on features indices, and a possibility to use column names added as a "nice-to-have"?

I really like @hcho3 idea of adapting the dmlc::JSONReader to handle the input. We would be able to pass the features indices directly, and also wouldn't need the nint_constraints parameter.

E.g., an easy way to convert names to ids:
```R
cols2ids <- function(object, col_names) {
  LUT <- seq_along(col_names) - 1
  names(LUT) <- col_names
  rapply(object, function(x) LUT[x], classes="character", how="replace")
}
param = list(letters[1:4], letters[11:15])
str(cols2ids(param, letters))
```

Reply:
@khotilov, @hcho3 Im happy to make changes to based on the dmlc::JSONReader idea.  I think I can handle changes to the R logic and the C++ logic (once the values have been read into C++).  But I have no clue on how to make C++ use dmlc::JSONReader so would definitely need help on that.